### DISCUSSION TITLE
Interesting other VFSes

### DISCUSSION THREAD

**Discussion by daenney** [ncruces/go-sqlite3#117] [Ideas]
I came across https://jtarchie.com/posts/2024-07-02-optimizing-large-scale-openstreetmap-data-with-sqlite today which had this interesting bit of using Zstd's seekable format to implement a (read-only) VFS with compression.

Their code is here https://github.com/jtarchie/sqlitezstd which uses https://github.com/psanford/sqlite3vfs (together with the mattn Cgo driver).

That repository also has an interesting other VFS linked, one that allows (partially) retrieving SQLite databases over HTTP (for example when a DB is stored in S3). That also reminded me a bit of Steampipe but that seems to be a touch more involved.

**Comment by ncruces** [comment]
For zstandard, see: https://github.com/ncruces/go-sqlite3/issues/105

I _think_ (hope?) that I don't need to do anything here. You should be able just use the [reader VFS](https://pkg.go.dev/github.com/ncruces/go-sqlite3/vfs/readervfs) with [zstd-seekable-format-go](https://github.com/SaveTheRbtz/zstd-seekable-format-go). If you can try and report back, it'd be very welcome. I could add an example to the docs.

**Comment by ncruces** [comment]
Steampipe is actually a (collection of) virtual table(s), not a VFS.

 I [really wanted](https://github.com/ncruces/go-sqlite3/pull/56) to get them working, but they [don't seem interested](https://github.com/turbot/steampipe-plugin-sdk/issues/741).

Sadly, it doesn't really help their business model so I doubt they ever will.

**Comment by ncruces** [comment]
For HTTP I already have a working (testable!) example:
https://github.com/ncruces/go-sqlite3/blob/fff8b1c74f961653f9717704a63cfa83864260ee/vfs/readervfs/example_test.go#L21-L66
